pyvrp/tests/test_CostEvaluator.py

```diff
@@ -58,3 +60,38 @@ def test_tw_penalty():
     assert_equal(cost_evaluator.tw_penalty(0), 0)
     assert_equal(cost_evaluator.tw_penalty(1), 4)
     assert_equal(cost_evaluator.tw_penalty(2), 8)
+
+
+@mark.parametrize(
+    "routes",
+    [[[1, 2], [3], [4]], [[1, 2, 3, 4]]],
+)
+def test_cost_and_penalized_cost(routes):
+    data = read("data/OkSmall.txt")
+    penalty_capacity = 20
+    penalty_tw = 6
+    default_cost_evaluator = CostEvaluator()
+    cost_evaluator = CostEvaluator(penalty_capacity, penalty_tw)
+
+    individual = Individual(data, routes)
+
+    load_penalty_cost = penalty_capacity * individual.excess_load()
+    tw_penalty_cost = penalty_tw * individual.time_warp()
+    distance = individual.distance()
+
+    # Test penalised cost
+    expected_cost = distance + load_penalty_cost + tw_penalty_cost
+    assert_equal(cost_evaluator.penalised_cost(individual), expected_cost)
+
+    # Default cost evaluator has 0 weights and only computes distance as cost
+    assert_equal(default_cost_evaluator.penalised_cost(individual), distance)
+
+    # Test normal cost, should be distance or infinite if infeasible
+    if individual.is_feasible():
+        assert_equal(cost_evaluator.cost(individual), distance)
+        assert_equal(default_cost_evaluator.cost(individual), distance)
+    else:
+        # C++ code represents infinite as max value for unsigned integer
+        INFEAS_COST = np.iinfo(np.uint32).max
+        assert_equal(cost_evaluator.cost(individual), INFEAS_COST)
+        assert_equal(default_cost_evaluator.cost(individual), INFEAS_COST)
```

Comment:
Can we separate this test the feasible and infeasible individual? It's not clear to me which of the individuals are feasible (and whether we hit all the assertions in the conditionals).

Reply:
I think it'd also be easier to read if the test is divided into two sections, where one section uses the default cost evaluator, and one uses the non-default one.

Reply:
I separated into two tests (for `cost` and `penalised_cost`) with both individuals instantiated inside the test.